public/js/dashboardCategories.js

```diff
@@ -68,6 +68,56 @@ window.deleteCategory = function deleteCategory(categoryID) {
   console.log('deleted' + categoryID);
 };
 
+const inputBoxTemplate = `
+  <div id="{{instructionID}}">
+    <input id="{{inputID}}" name="{{inputID}}" type="text"></input>
+    <button id="{{removeButtonID}}" onclick="removeInstructionInput(this.id)" hidden>-</button>
+    <button id="{{addButtonID}}" onclick="addInstructionInput(this.id)">+</button><br>
+  </div>
+`;
+
+/**
+ * Adds an input box to the correct section of the page.
+ * @param {string} prevInstructionID ID of the previous input
+ * box's add button [instructionType]-add-[instructionIndex].
+ */
+function addInstructionInput(prevInstructionID) {
+  const prevInstructionIdArr = prevInstructionID.split('-');
+  const instructionType = prevInstructionIdArr[0];
+  const currInstructionIndex = parseInt(prevInstructionIdArr[2]) + 1;
+
+  const prevRemoveButton = document.getElementById(
+    `${instructionType}-remove-${prevInstructionIdArr[2]}`
+  );
+  if (prevRemoveButton) prevRemoveButton.hidden = false;
+
+  const prevAddButton = document.getElementById(
+    `${instructionType}-add-${prevInstructionIdArr[2]}`
+  );
+  if (prevAddButton) prevAddButton.hidden = true;
+
+  const renderInputBox = Handlebars.compile(inputBoxTemplate);
+  document.getElementById(`${instructionType}-container`).innerHTML += renderInputBox({
+    instructionID: `${instructionType}-container-${currInstructionIndex}`,
+    inputID: `${instructionType}-input-${currInstructionIndex}`,
+    removeButtonID: `${instructionType}-remove-${currInstructionIndex}`,
+    addButtonID: `${instructionType}-add-${currInstructionIndex}`,
```

Comment:
It looks like all of these parameters can be derived from ```instructionType``` and ```currInstructionIndex```. Could we just code the template to take two params?